x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/json_diff/json_diff.test.tsx

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { render, within, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import type { Matcher } from '@testing-library/react';
+import { escapeRegExp } from 'lodash';
+
+import { RuleDiffTab } from '../rule_diff_tab';
+import { savedRuleMock } from '../../../logic/mock';
+import type { RuleResponse } from '../../../../../../common/api/detection_engine/model/rule_schema/rule_schemas.gen';
+
+/**
+ * Creates a matcher function designed to determine if an element contains specific text content in specific order.
+ * It returns true if the element itself contains the strings (matching all strings in the specified order) without
+ * any of its children containing all of these strings. This is useful for finding elements with text content that
+ * is split across multiple child elements.
+ *
+ * @param {string[]} strings - An array of strings to be matched within the content.
+ * @returns {Function} A Matcher function that returns a boolean indicating whether the element matches the
+ * specified text conditions: the element contains the text specified by `strings` and none of its children
+ * contain this text.
+ *
+ * The Matcher function itself has the following signature:
+ * @param {string} content - The content string to be checked. Not directly used in the current implementation
+ * but is required for the matcher's signature.
```

Comment:
If I understand correctly, `content` needs to be passed here cause we need to return a function that is passed either to `getByText` and `queryAllByText`, and both those function require that signature. 

Took me a bit to understand, maybe add this additional detail in this comment?

Reply:
You're right, it's too complicated. So I refactored a bit and removed this helper altogether.